### ISSUE TITLE
Discussion: How to reach UI/UX decision makers at Microsoft to discuss Universal Keybindings?

### ISSUE THREAD

**Issue by ylluminate** [microsoft/microsoft-ui-xaml#1119]
How can I get in touch with folks at Microsoft who make decisions about keybindings and such UI/UX issues?  I need to have a conversation with them...

As a long term Windows AND macOS (and *nix) user (over 30 years), I have some important feedback that has never seemed to reach anyone through normal feedback channels.

Apple arguably has most cohesive and well-defined user experience / UI model anywhere at the moment. Their work to standardize the user interface and user interaction has been top notch since the days of NeXT. One area that this cohesion has influenced greatly is keyboard interaction / keybindings.

As a developer I abhor the keybindings that Windows has and has had over the decades. They've always felt very forced and unnatural. This becomes greatly pronounced after you use macOS.

Microsoft desperately needs to have a complete and comprehensive keybindings option that would remap various keys and key combinations to those that are used in macOS. This is especially poignant for developers and text editors / writers. Simple things like cmd(control?)-a, cmd-e, cmd-left-arrow/right-arrow, option(alt)-delete or the plethora of other options make using macOS much more of a joy than any other operating system.

People can argue that it's just what you're used to and that folks can learn new things but: A) having used Windows longer than macOS I can call bull crap on this, and B) why force people into using a set of traditional Windows keybindings and combos when they might be coming from macOS or Emacs?

This is a severely critical issue and frankly I've talked to many hundreds of people who laugh at Microsoft and joke around about how they simply could never never convert to Windows due to the keybindings alone.

Why not just remove this hurdle and make the possibility to switch OR use both operating systems as many of us do a more cohesive preference.

You've gone ahead and caved on the dark theme issue so now let's move onto some other important and progressive concepts such as "Universal Keybindings" that move with a user from OS to OS no matter what OS they're using!

**Comment by MarkIngramUK** [comment]
This is a decision for application developers themselves, Microsoft does not force keybindings on developers (though they do issue [guidance](https://docs.microsoft.com/en-us/previous-versions/windows/desktop/dnacc/guidelines-for-keyboard-user-interface-design)).

**Comment by ylluminate** [comment]
I can't see that as being true for general text editing and system tools/utilities. So there's no way to assign via the operating system that <cmd-e> also acts as the <end> key?

**Comment by jevansaks** [comment]
@chigy and others put a lot of time into the keyboarding guidance docs: https://docs.microsoft.com/en-us/windows/uwp/design/input/keyboard-interactions

@ylluminate assuming you agree with those keybindings (or please call out any important ones you think are missing) then is the problem just that across Windows the different apps/controls don't consistently respect them?

Personally as a heavy user of "ctrl+backspace" in text boxes to delete words it was frustrating to me that Notepad and Win32 edit boxes didn't do this right until Windows 1903, so I hear you!

**Comment by chigy** [comment]
Hello @ylluminate,
Thank you for your feedback. Let me try to understand just exactly what you are suggesting we consider. For one, I would like to understand better what you are asking for Windows keybindings to do differently. As you may be aware, there are multiple causes that Windows keybindings might not do what exactly you want to do but I must first understand what you want which I am not completely clear yet. So can we start from there?

Are you asking any of the following or combination of them?
- Windows do not support a well defined universal key combination like cmd-e.
- Windows apps do not support universal key combination consistently.
- Windows should provide systematic way to replace Windows key combination to what user is used to using with macOS.
- Something else?

Thank you for your input in advance!

**Comment by chingucoding** [comment]
@ylluminate Friendly ping.

**Comment by Gavin-Williams** [comment]
I think he is just wanting the ability to customize hot-keys? So dot-point 3 on chigy's list.

**Comment by ylluminate** [comment]
Hey thanks to @chigy, @jevansaks, @MarkIngramUK, @Gavin-Williams, and @chingucoding for engaging in this discussion. Your insights, and the work you do for Windows, are much appreciated. Apologies for letting this languish - I think life (for all of us) has been a bit crazy the last few years and this disappeared from my horizon since I ended up moving forward with Apple Silicon and macOS vs making the full transition to Windows as I had contemplated.

To clarify my position and address your queries, let's break down the matter into two key issues:

1. **Consistency across Applications**: This is a matter of having a universal set of keybindings that works seamlessly across Windows applications. Currently, keybindings tend to differ between applications, which can create a disjointed user experience.

2. **Customizability of Keybindings**: Users coming from different OSs, such as macOS or Linux, have become accustomed to a certain set of keybindings. When transitioning to Windows, the inability to customize keybindings can prove to be a hurdle. For such users, and even long-time Windows users, the ability to customize keybindings (including the option to mirror those of other OSs) could greatly enhance the user experience.

The hope is for Windows to support a system-wide keybinding scheme that can be customized according to user preference. This includes supporting keybindings that closely mimic those of other OSs, hence making the transition between different operating systems smoother. Ideally, users should also be able to save and load these keybinding profiles, providing further convenience and flexibility.

While I understand that individual application developers have some control over their keybindings, having OS-level support for keybindings could drive consistency across applications and facilitate a more universal Windows user experience. 

Hence, my suggestion aligns with the third point on @chigy's list, but with added emphasis on consistency across applications.

I hope this clarification helps, and I'm open to further discussions. 

Yes, this could be a complex issue, but I think collective deliberation and attempts to address this could lead to a really nice outcome if we can make it work.
